### PR TITLE
Fix Crystal Geometry Cppcheck warnings

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR fixes some cppcheck warnings in the Crystal Geometry targets.

**To test:**
Code review and check tests pass.

*There is no associated issue.*

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/CurveFitting/src/Functions/CrystalFieldMultiSpectrum.cpp [hunk 1]
```diff
@@ -122,15 +122,13 @@ void CrystalFieldMultiSpectrum::init() {
 }
 
 size_t CrystalFieldMultiSpectrum::getNumberDomains() const {
-  if (!m_target) {
+  if (!m_target)
```
Comment (ConorMFinn) [inline Framework/CurveFitting/src/Functions/CrystalFieldMultiSpectrum.cpp:125]: Is removing the braces a requirement of CppCheck? David and I had a discussion about these kinds of `if` statements as they can make it easier to introduce bugs. There's quite an interesting post about it where using these caused a security flaw in an old version of OSX, the "goto fail" bug: https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/

**Review by cailafinn — CHANGES_REQUESTED** [review]
Just one quick note on some if changes, otherwise looks good

File: Framework/CurveFitting/src/Functions/CrystalFieldMultiSpectrum.cpp (hunk 1 above)
Comment (robertapplin) [reply Framework/CurveFitting/src/Functions/CrystalFieldMultiSpectrum.cpp:125]: No its not a requirement of cppcheck, I'll add them back in. Thats an interesting read, thanks!

**Comment by robertapplin** [comment]
Changed target to `release-next` as recommended by @thomashampson

**Review by cailafinn — DISMISSED** [review]
LGTM, nice work 👍

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: Framework/Crystal/src/PeakHKLErrors.cpp [hunk 2]
```diff
@@ -163,21 +163,20 @@ void PeakHKLErrors::cLone(std::shared_ptr<Geometry::ParameterMap> &pmap,
...
-  auto pmap = std::make_shared<Geometry::ParameterMap>();
+std::shared_ptr<Geometry::Instrument> PeakHKLErrors::getNewInstrument(const PeaksWorkspace_sptr &peaksWs) const {
+  Geometry::Instrument_const_sptr instSave = peaksWs->getPeak(0).getInstrument();
 
   if (!instSave) {
     g_log.error(" Peaks workspace does not have an instrument");
     throw std::invalid_argument(" Not all peaks have an instrument");
   }
 
+  auto pmap = std::make_shared<Geometry::ParameterMap>();
```
[the remark this answers is not in the document]
Comment (robertapplin) [reply Framework/Crystal/src/PeakHKLErrors.cpp:179]: Its used further down. I moved it a few lines so it was below the check on `instSave`

File: Framework/Crystal/src/IndexSXPeaks.cpp [hunk 3]
```diff
@@ -204,6 +204,8 @@ void IndexSXPeaks::exec() {
   }
   prog.report(); // 1st Progress report.
 
+  assert(peaks.size() >= 2);
```
[the remark this answers is not in the document]
Comment (martyngigg) [reply Framework/Crystal/src/IndexSXPeaks.cpp:207]: Ah okay. Is there a reason we need anything at all then? Was this something that cppcheck warned about?

[the remark this answers is not in the document]
Comment (robertapplin) [reply Framework/Crystal/src/IndexSXPeaks.cpp:207]: Sounds good, I'll make those changes

**Comment by robertapplin** [comment]
I've changed the assert statement back to `>= 2` because using `== npeaks` doesn't fix the cppcheck warning

**Review by cailafinn — DISMISSED** [review]

**Comment by robertapplin** [comment]
Not sure where we got to with this, I think @ConorMFinn approved it before the base branch was changed to `main`. @martyngigg are you happy to take another quick look?

**Review by cailafinn — DISMISSED** [review]
Still looks good to me after the base branch change, and Martyn's comments have been accounted for

**Review by martyngigg — APPROVED** [review]

**Pull request merged**